dune_api_scripts/store/solver_list.py

```diff
@@ -0,0 +1,62 @@
+"""Modifies and executed dune query for today's data"""
+from __future__ import annotations
+
+import dataclasses
+import json
+import os
+from dataclasses import dataclass
+from typing import Any
+
+from duneapi.api import DuneAPI
+from duneapi.types import DuneQuery, Network
+from duneapi.types import Address
+
+
+class EnhancedJSONEncoder(json.JSONEncoder):
+    def default(self, o):
+        if dataclasses.is_dataclass(o):
+            return dataclasses.asdict(o)
+        return super().default(o)
+
+
+@dataclass
+class Solver:
+    address: str
+    environment: str
+    name: str
+    active: bool
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> Solver:
+        return cls(
+            address=Address(data["address"]).address,
+            environment=data["environment"],
+            name=data["name"],
+            active=data["active"],
+        )
+
+
+def store_solver_list(dune: DuneAPI) -> list[Solver]:
+    # TODO - fetch for both networks and merge JSON content with chainID
+    # Waiting on https://github.com/duneanalytics/abstractions/pull/1268
+    raw_solver_list = dune.fetch(
+        DuneQuery.from_environment(
+            name="Solver List",
+            description="",
+            raw_sql="select * from gnosis_protocol_v2.view_solvers",
+            network=Network.MAINNET,
+            parameters=[],
+        )
+    )
+    solver_list = [Solver.from_dict(rec) for rec in raw_solver_list]
+
+    filename = os.path.join(os.environ["DUNE_DATA_FOLDER"], "solvers.json")
+    with open(filename, "w+", encoding="utf-8") as f:
+        json.dump(solver_list, f, ensure_ascii=False, indent=4, cls=EnhancedJSONEncoder)
```

Comment:
Just to clarify if I'm not missing anything.

There's no transformation in the data received from Dune at all, it's stored pretty much verbatim from what's received.
Then, the dataclass and EnhancedJSONEncoder are not really needed, you could just store that in a file directly, correct?

Of course, they serve as documentation and to validate the data received, but I just want to make sure there's no hidden need that I'm not seeing here.

Reply:
Dataclasses are not JSON serializable by default, this `EnhancedJSONEncoder` is a generic solution to serialize any dataclass as JSON by using its inherited `asdict`function. Without this we would get an error `object of type Solver is not JSON serializable`. The nice thing here is that `EnhancedJSONEncoder` appears to be a universal solution for serializing any dataclass object  (haven't yet tested how it works on nested structures).

Reply:
BTW, I got this from here (first answer)

https://stackoverflow.com/questions/51286748/make-the-python-json-encoder-support-pythons-new-dataclasses

Reply:
I got the serialization part.
My point was, other than documentation, you could just save the JSON returned from the query and be done with it. Or is there any other hidden logic that I am not seeing?

Reply:
Right of course! Silly me. Yea, so two things that I would have to check first.

1. Addresses come from dune in the form `\x....` instead of being prefixed with `0x` and they are lower case. I can adapt the query to return `0x` prefixed address directly, but it would still be lower case.
2. Just gotta see what type the "active" field is returned as. If it is actually a `json` boolean field then we should be golden. 

Now that you mention it, I would also prefer to write the response directly to JSON without the need for any intermediary tampering. As long as we are all ok with lower case addresses. This would eliminate the special encoding and the need for a data class at all. Making the entire script much smaller!

Reply:
All done! Check it out.